feat: send telemetry when creating new provider connection closed/canceled

dgolovin opened the pull request:
### What does this PR do?

It adds telemetry reporting when 'Cancel' pressed in 'Create New Provider Connection' form 

### Screenshot/screencast of this PR

N/A

### What issues does this PR fix or reference?

https://github.com/redhat-developer/podman-desktop-sandbox-ext/issues/51

### How to test this PR?

<!-- Please explain steps to reproduce -->

lstocchi reviewed (approved):
LGTM

dgolovin commented:
I also added telemetry for 'Close' button when form just closed without cancelling anything.

benoitf reviewed (changes requested):
there is no unit test that is checking that we receive the events being added

I also think here that the name of the event could be more accurate to differentiate when we're aborting a display thing like 'dialog'/'page' and when aborting the process of creating a connection

so it could match new event title of https://github.com/containers/podman-desktop/pull/3697

`createNewProviderConnectionPageClosed` (or other names where we're sure that we're closing a display thing) `createNewProviderConnectionPageCloseRequest`, etc.

I would also use the event as a suffix like `createNewProviderConnectionUserCancelled`

benoitf commented:
@dgolovin if you can add a test checking those events and I think we're good to merge 👍

dgolovin commented:
Tests are added, but meantime I found couple issues:
1. I added telemetry for close button to close progress panel not the page itself (fixed)
2.  close function shadows window.close, so I did a little refactoring close -> closePanes and added closePage with router.goto and telemetry calls

benoitf reviewed (approved):
Glad you found issues while testing the component 🎉
